securemailbox/views/retrieve.py

```diff
@@ -1,11 +1,78 @@
-from flask import Blueprint, jsonify
+from flask import Blueprint, jsonify, request, json
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound
+
+from ..models import Mailbox
+from ..models import Message
 
 from securemailbox import db
 
 # Create the `retrieve` blueprint
 retrieve_blueprint = Blueprint("retrieve", __name__)
 
-
 @retrieve_blueprint.route("/retrieve/", methods=["POST"])
 def retrieve():
-    return jsonify({"success": True, "error": None})
+
+    # Check for valid json request
+    if not request.is_json:
+        return jsonify({'success': False, 'error': 'Request must be valid json', 'data': None}), 400
```

Comment:
Nit: You use single quotes someplaces and double quotes other places. I think the black formatting tool will automagically fix this for you.

Reply:
just ran black, thanks for the reminder